frame/child-bounties/src/benchmarking.rs

```diff
@@ -0,0 +1,302 @@
+	let curator_lookup = T::Lookup::unlookup(bounty_setup.curator.clone());
+	Bounties::<T>::propose_bounty(
+		RawOrigin::Signed(bounty_setup.caller.clone()).into(),
+		bounty_setup.value,
+		bounty_setup.reason.clone(),
+	)?;
+
+	bounty_setup.bounty_id = Bounties::<T>::bounty_count() - 1;
+
+	Bounties::<T>::approve_bounty(RawOrigin::Root.into(), bounty_setup.bounty_id)?;
+	Treasury::<T>::on_initialize(T::BlockNumber::zero());
+	Bounties::<T>::propose_curator(
+		RawOrigin::Root.into(),
+		bounty_setup.bounty_id,
+		curator_lookup.clone(),
+		bounty_setup.fee,
+	)?;
+	Bounties::<T>::accept_curator(
+		RawOrigin::Signed(bounty_setup.curator.clone()).into(),
+		bounty_setup.bounty_id,
+	)?;
+
+	Ok(bounty_setup)
+}
+
+fn create_child_bounty<T: Config>(u: u32, d: u32) -> Result<BenchmarkChildBounty<T>, &'static str> {
+	let mut bounty_setup = create_bounty::<T>(u, d)?;
+	let child_curator_lookup = T::Lookup::unlookup(bounty_setup.child_curator.clone());
+
+	ChildBounties::<T>::add_child_bounty(
+		RawOrigin::Signed(bounty_setup.curator.clone()).into(),
+		bounty_setup.bounty_id,
+		bounty_setup.child_bounty_value,
+		bounty_setup.reason.clone(),
+	)?;
+
+	bounty_setup.child_bounty_id = ChildBountyCount::<T>::get() - 1;
+
+	ChildBounties::<T>::propose_curator(
+		RawOrigin::Signed(bounty_setup.curator.clone()).into(),
+		bounty_setup.bounty_id,
+		bounty_setup.child_bounty_id,
+		child_curator_lookup.clone(),
+		bounty_setup.child_bounty_fee,
+	)?;
+
+	ChildBounties::<T>::accept_curator(
+		RawOrigin::Signed(bounty_setup.child_curator.clone()).into(),
+		bounty_setup.bounty_id,
+		bounty_setup.child_bounty_id,
+	)?;
+
+	Ok(bounty_setup)
+}
+
+fn setup_pot_account<T: Config>() {
+	let pot_account = Bounties::<T>::account_id();
+	let value = T::Currency::minimum_balance().saturating_mul(1_000_000_000u32.into());
+	let _ = T::Currency::make_free_balance_be(&pot_account, value);
+}
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+benchmarks! {
+	add_child_bounty {
+		let d in 0 .. MAX_BYTES;
+		setup_pot_account::<T>();
+		let bounty_setup = create_bounty::<T>(0, d)?;
+	}: _(RawOrigin::Signed(bounty_setup.curator), bounty_setup.bounty_id,
+			bounty_setup.child_bounty_value, bounty_setup.reason.clone())
+	verify {
+		assert_last_event::<T>(Event::ChildBountyAdded {
+			index: bounty_setup.bounty_id,
+			child_index: bounty_setup.child_bounty_id,
+		}.into())
+	}
+
+	propose_curator {
+		setup_pot_account::<T>();
+		let mut bounty_setup = create_bounty::<T>(0, MAX_BYTES)?;
+		let child_curator_lookup = T::Lookup::unlookup(bounty_setup.child_curator.clone());
+
+		ChildBounties::<T>::add_child_bounty(
+			RawOrigin::Signed(bounty_setup.curator.clone()).into(),
+			bounty_setup.bounty_id,
+			bounty_setup.child_bounty_value,
+			bounty_setup.reason.clone(),
+		)?;
+		bounty_setup.child_bounty_id = ChildBountyCount::<T>::get() - 1;
```

Comment:
see it as a nit but I am not sure if you need this `bounty_setup` at all. Some fields seem to never be used. Sometimes like here, there's not much point in assigning values to a field and reading it in the next like. You might as well pass `ChildBountyCount::<T>::get() - 1` directly to the benchmark below.